crates/core/src/json_pointer.rs

```diff
@@ -65,14 +61,17 @@ impl JsonPointer {
     /// # Safety
     ///
     /// The input string *must* be a valid JSON pointer.
-    pub const unsafe fn new_unchecked(s: &[u8]) -> &Self {
+    pub const unsafe fn new_unchecked(s: &str) -> &Self {
```

Comment:
Please make sure this accepts a `&[u8]`

Reply:
... and I've added a `new_unchecked_str` to match.